socorro/processor/mozilla_transform_rules.py

```diff
@@ -770,46 +771,68 @@ def _get_real_version(self, product, build_id, channel):
 
         session = session_with_retries(self.buildhub_api)
 
-        query = {
-            'source.product': product,
-            'build.id': '"%s"' % build_id,
-            'target.channel': channel,
+        es_query = {
+            'query': {
+                'bool': {
+                    'must': {
+                        'match_all': {}
+                    },
+                    'filter': [
+                        {
+                            'term': {
+                                'source.product': product.lower()
+                            }
+                        },
+                        {
+                            'term': {
+                                'build.id': str(build_id)
+                            }
+                        },
+                        {
+                            'term': {
+                                'target.channel': channel
+                            }
+                        },
+                    ],
+                }
+            },
+            'size': 1
         }
 
         if channel == 'release':
-            # If it's the release channel, we can specify we only want to see
-            # "rc" versions and thus limit the result set to 1
-            query['like_target.version'] = '*rc*'
-            query['_limit'] = 1
+            # If it's the release channel, we only want to see rc versions
+            es_query['query']['bool']['filter'].append(
+                {
+                    'wildcard': {
+                        'target.version': '*rc*'
+                    }
+                }
+            )
+        else:
+            # If it's the beta channel, we want to exclude rc versions
+            es_query['query']['bool']['must_not'] = {
+                'wildcard': {
+                    'target.version': '*rc*'
+                }
+            }
 
-        resp = session.get(self.buildhub_api, params=query)
+        resp = session.post(self.buildhub_api, data=json.dumps(es_query, sort_keys=True))
```

Comment:
We sort the keys here since it makes it easier to test. It's a small dict, so I think the performance hit is pretty meh.